program/tests/helpers/common.rs

```diff
@@ -0,0 +1,79 @@
+use {
+    crate::helpers::mint_builder::TokenProgram,
+    mollusk_svm::Mollusk,
+    solana_account::Account,
+    solana_program_option::COption,
+    solana_program_pack::Pack,
+    solana_pubkey::Pubkey,
+    solana_rent::Rent,
+    spl_pod::{
+        optional_keys::OptionalNonZeroPubkey,
+        primitives::{PodBool, PodU64},
+    },
+    spl_token_2022::{
+        extension::{
+            mint_close_authority::MintCloseAuthority, BaseStateWithExtensionsMut, ExtensionType,
+            PodStateWithExtensionsMut,
+        },
+        pod::{PodCOption, PodMint},
+    },
+    std::convert::TryFrom,
+};
+
+pub fn init_mollusk() -> Mollusk {
+    let mut mollusk = Mollusk::new(&spl_token_wrap::id(), "spl_token_wrap");
+    mollusk_svm_programs_token::token::add_program(&mut mollusk);
+    mollusk_svm_programs_token::token2022::add_program(&mut mollusk);
+    mollusk
+}
+
+pub const MINT_DECIMALS: u8 = 12;
+pub const MINT_SUPPLY: u64 = 500_000_000;
+pub const FREEZE_AUTHORITY: Pubkey =
+    Pubkey::from_str_const("11111115q4EpJaTXAZWpCg3J2zppWGSZ46KXozzo9");
+
+fn token_2022_with_extension_data(supply: u64) -> Vec<u8> {
+    let mint_size =
+        ExtensionType::try_calculate_account_len::<PodMint>(&[ExtensionType::MintCloseAuthority])
```

Comment:
To test the behavior when the token account has an extension, you can add the `TransferFeeConfig` extension to the mint